src/status_im/ui/screens/contacts/events.cljs

```diff
@@ -346,6 +346,12 @@
                     [:watch-contact contact']
                     [:discoveries-send-portions chat-or-whisper-id]]})))
 
+(register-handler-fx
+  :add-pending-contact-and-open-chat
+  (fn [_ [_ whisper-id]]
+    {:dispatch-n [[:add-pending-contact whisper-id]
+                  [:start-chat whisper-id {:navigation-replace? true}]]}))
```

Comment:
Please, instead of delegating dispatched in this (now atomic handler), compose the function instead.
For starting chat, the function producing necessary re-frame effects is here - https://github.com/status-im/status-react/blob/develop/src/status_im/chat/events.cljs#L298-L306 (it's anonymous function currently, , now it's right time to make it named and re-use it :) )
For adding pending contact, it's here - https://github.com/status-im/status-react/blob/develop/src/status_im/ui/screens/contacts/events.cljs#L337-L347 (anonymous function again, and worse then starting chat, because its unnecessary dispatching events instead of composing functions).
See this for rationale - https://github.com/status-im/ideas/issues/31#issuecomment-346345981 (point 3.)

Reply:
and also please check you should go back to home from chat screen not to add new contact

Reply:
@janherich 
just to make sure we are on the same page, you mean something like:

https://github.com/cammellos/status-react/commit/bb8c6f06dd44bf302263a4f542d1e249d608234c#diff-0da314cfa3a44d32c27ef25c73fca7cbR320

eventually replacing the last :dispatch as well (::send-contact-request)

Reply:
@cammellos Exactly !